server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -127,6 +183,55 @@ protected void checkHasIllegalOptions(List<?> rules, String options) {
     }
   }
 
+  /**
+   * Helper method to collect analysis errors if the rule context contains illegal options
+   *
+   * @param rule Generic rule to check
+   * @param options Options checked to insert into error message
+   */
+  protected void checkHasIllegalOptions(TerminalNode rule, String options) {
+    if (!rule.getText().isEmpty()) {
```

Comment:
This condition is just always true, isn't it?

Reply:
No, it can be of the type but not have any content/text associated with it. You see this with the `*` rules quite often.

Reply:
But this is a TerminalNode - it is created to represent a matched token.

Reply:
This function has been removed in the current commit. It was a lesser form of the checkPrerequisiteIsMet function.